[LOW] Refactor endTurn() in engine.ts - cleanup responsibilities should be separated

## Summary

The `endTurn()` method in `src/engine.ts:1501-1534` contains 34 lines of turn cleanup logic including flag resets, monster returns, hand limit enforcement, and AI turn triggering. While the method is primarily orchestrating other methods, it contains embedded logic for state transitions.

**Location:** `src/engine.ts:1501-1534` (34 lines)

## Impact

- **Cyclomatic complexity**: Estimated 7+ (moderate, but combined with advancePhase creates duplication)
- **Duplication**: Similar cleanup logic exists in `advancePhase()` - violation of DRY
- **Mixed concerns**: Cleanup logic mixed with turn transition and AI scheduling
- **AI coupling**: Direct setTimeout and aiTurn call creates tight coupling

## Evidence

```typescript
endTurn(){
  this._resetMonsterFlags('player');
  this._returnTempStolenMonsters('player');
  this._returnSpiritMonsters('player');
  this._tickTurnCounters('player');

  this.state.player.normalSummonUsed   = false;
  this.state.opponent.normalSummonUsed = false;

  const hand = this.state.player.hand;
  while(hand.length > GAME_RULES.handLimitEnd){ hand.shift(); }

  this.state.activePlayer = 'opponent';
  this.state.phase = 'draw';
  this.state.turn++;

  this.addLog(`=== Round ${this.state.turn} - Opponent's turn ===`);
  this.ui.render(this.state);

  setTimeout(() => {
    aiTurn(createEngineDependencies(this)).catch(err => {
      EchoesOfSanguo.log('ERROR', 'AI turn crashed:', err);
      EchoesOfSanguo.downloadLog('ai_crash');
      this.state.activePlayer = 'player';
      this.state.phase = 'main';
      this.state.turn++;
      this.state.oneMoveActionUsed = false;
      this.addLog(`[ERROR] Opponent AI crashed. Your turn (Round ${this.state.turn}).`);
      this.refillHand('player');
      this.ui.render(this.state);
    });
  }, 600);
}
```

**Issues:**
- 4 cleanup method calls (could be in a loop or single method)
- Duplicate flag resets for both players
- Hand limit enforcement duplicated from advancePhase pattern
- AI scheduling with inline error handling (20+ lines)

## Recommended Fix

1. **Consolidate cleanup to single method:**
   ```typescript
   private _cleanupForTurnEnd(owner: Owner): void {
     this._resetMonsterFlags(owner);
     this._returnTempStolenMonsters(owner);
     this._returnSpiritMonsters(owner);
     this._tickTurnCounters(owner);
     this._resetSummonFlags(owner);
     this._enforceHandLimit(owner);
   }
   ```

2. **Extract AI turn scheduling:**
   ```typescript
   private _scheduleAITurn(delayMs: number = 600): void {
     setTimeout(() => {
       aiTurn(createEngineDependencies(this))
         .catch(err => this._handleAICrash(err));
     }, delayMs);
   }
   ```

3. **Simplify to orchestration:**
   ```typescript
   endTurn() {
     this._cleanupForTurnEnd('player');
     this._cleanupForTurnEnd('opponent');
     
     this._transitionToOpponentTurn();
     this._scheduleAITurn();
   }
   ```

**Estimated effort:** 25 minutes

## References

- Related: `advancePhase()` at `src/engine.ts:1397-1427` - shares cleanup logic
- Similar cleanup pattern: `_resetMonsterFlags()`, `_returnTempStolenMonsters()` at `src/engine.ts:1428-1499`

--- a/src/engine.ts
+++ b/src/engine.ts
@@ -1462,15 +1462,10 @@ export class GameEngine {
       this.addLog(`--- ${names[this.state.phase] ?? this.state.phase} ---`);
       this.ui.render(this.state);
     } else {
-      this._resetMonsterFlags(this.state.activePlayer);
-      this._returnTempStolenMonsters(this.state.activePlayer);
-      this._returnSpiritMonsters(this.state.activePlayer);
-      this._tickTurnCounters(this.state.activePlayer);
-      this.state[this.state.activePlayer].normalSummonUsed = false;
-      const opp = this.state.activePlayer === 'player' ? 'opponent' : 'player';
-      this.state[opp].normalSummonUsed = false;
-      const hand = this.state[this.state.activePlayer].hand;
-      while(hand.length > GAME_RULES.handLimitEnd){ hand.shift(); }
+      const active = this.state.activePlayer;
+      const opp = active === 'player' ? 'opponent' : 'player';
+      this._cleanupForTurnEnd(active);
+      this._cleanupForTurnEnd(opp);
       this.endTurn();
     }
   }
@@ -1548,38 +1543,57 @@ export class GameEngine {
     }
   }
 
-  endTurn(){
-    this._resetMonsterFlags('player');
-    this._returnTempStolenMonsters('player');
-    this._returnSpiritMonsters('player');
-    this._tickTurnCounters('player');
+  _cleanupForTurnEnd(owner: Owner): void {
+    this._resetMonsterFlags(owner);
+    this._returnTempStolenMonsters(owner);
+    this._returnSpiritMonsters(owner);
+    this._tickTurnCounters(owner);
+    this._resetSummonFlags(owner);
+    this._enforceHandLimit(owner);
+  }
 
-    this.state.player.normalSummonUsed   = false;
-    this.state.opponent.normalSummonUsed = false;
+  _resetSummonFlags(owner: Owner): void {
+    this.state[owner].normalSummonUsed = false;
+  }
 
-    const hand = this.state.player.hand;
-    while(hand.length > GAME_RULES.handLimitEnd){ hand.shift(); }
+  _enforceHandLimit(owner: Owner): void {
+    const hand = this.state[owner].hand;
+    while (hand.length > GAME_RULES.handLimitEnd) { hand.shift(); }
+  }
 
+  _handleAICrash(err: any): void {
+    EchoesOfSanguo.log('ERROR', 'AI turn crashed:', err);
+    EchoesOfSanguo.downloadLog('ai_crash');
+    this.state.activePlayer = 'player';
+    this.state.phase = 'main';
+    this.state.turn++;
+    this.state.oneMoveActionUsed = false;
+    this.addLog(`[ERROR] Opponent AI crashed. Your turn (Round ${this.state.turn}).`);
+    this.refillHand('player');
+    this.ui.render(this.state);
+  }
+
+  _transitionToOpponentTurn(): void {
     this.state.activePlayer = 'opponent';
     this.state.phase = 'draw';
     this.state.turn++;
-
     this.addLog(`=== Round ${this.state.turn} - Opponent's turn ===`);
     this.ui.render(this.state);
+  }
 
+  _scheduleAITurn(delayMs: number = 600): void {
     setTimeout(() => {
-      aiTurn(createEngineDependencies(this)).catch(err => {
-        EchoesOfSanguo.log('ERROR', 'AI turn crashed:', err);
-        EchoesOfSanguo.downloadLog('ai_crash');
-        this.state.activePlayer = 'player';
-        this.state.phase = 'main';
-        this.state.turn++;
-        this.state.oneMoveActionUsed = false;
-        this.addLog(`[ERROR] Opponent AI crashed. Your turn (Round ${this.state.turn}).`);
-        this.refillHand('player');
-        this.ui.render(this.state);
-      });
-    }, 600);
+      aiTurn(createEngineDependencies(this))
+        .catch(err => this._handleAICrash(err));
+    }, delayMs);
+  }
+
+  endTurn(){
+    this._cleanupForTurnEnd('player');
+    this._cleanupForTurnEnd('opponent');
+    
+    this._transitionToOpponentTurn();
+    this._scheduleAITurn();
   }
 
   changePosition(owner: Owner, zone: number){

